src/GafferSceneUI/TransformTool.cpp

```diff
@@ -111,21 +111,27 @@ bool updateSelection( const SceneAlgo::History *history, TransformTool::Selectio
 
 	if( const ObjectSource *objectSource = runTimeCast<const ObjectSource>( node ) )
 	{
-		selection.transformPlug = const_cast<TransformPlug *>( objectSource->transformPlug() );
-		selection.transformSpace = M44f();
+		if( history->scene == objectSource->outPlug() )
+		{
+			selection.transformPlug = const_cast<TransformPlug *>( objectSource->transformPlug() );
+			selection.transformSpace = M44f();
+		}
 	}
 	else if( const Group *group = runTimeCast<const Group>( node ) )
 	{
 		const ScenePlug::ScenePath &path = history->context->get<ScenePlug::ScenePath>( ScenePlug::scenePathContextName );
-		if( path.size() == 1 )
+		if( history->scene == group->outPlug() && path.size() == 1 )
```

Comment:
Silly Q no.2 Are these new `outPlug()` tests because we now have in plugs in the history too?

Reply:
Yes. The same fix would also have been needed if someone added additional "out" plugs to a scene node, but it is the existence of "in" plugs in the history that makes the fix essential right now.